app/src/main/java/org/dhis2/data/event/DhisEventUtils.kt

```diff
@@ -0,0 +1,112 @@
+package org.dhis2.data.event
+
+import javax.inject.Inject
+import org.dhis2.utils.DateUtils
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.arch.helpers.UidsHelper
+import org.hisp.dhis.android.core.enrollment.EnrollmentStatus
+import org.hisp.dhis.android.core.event.Event
+import org.hisp.dhis.android.core.event.EventStatus
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
+
+class DhisEventUtils @Inject constructor(val d2: D2) {
+
+    fun isEventEditable(eventUid: String): Boolean {
+        val event =
+            d2.eventModule().events().uid(eventUid).blockingGet()
+        val program =
+            d2.programModule().programs().uid(event.program()).blockingGet()
+        val stage =
```

Comment:
I would go a bit further and have isEnrollmentOpen, inOrgUnitRange, getCatComboAccess etc.. in different classes.
Then you could easily add unit test :dancing_women: XD

Reply:
I have all different classes. The tests are too much to get this ready now. So i'll do it next week.